python/lsst/ts/observatory/control/auxtel/atcs.py

```diff
@@ -0,0 +1,1845 @@
+        self.log.debug("Done")
+
+    async def offset_radec(self, ra, dec):
+        """ Offset telescope in RA and Dec.
+
+        Parameters
+        ----------
+        ra : `float`
+            Offset in ra (arcsec).
+        dec : `float` or `str`
+            Offset in dec (arcsec).
+
+        """
+        self.log.debug(f"Applying RA/Dec offset: {ra}/ {dec} ")
+        await self.rem.atptg.cmd_offsetRADec.set_start(type=0, off1=ra, off2=dec, num=0)
+        self.log.debug("Waiting for telescope to settle.")
+        await asyncio.sleep(self.tel_settle_time)
+        self.log.debug("Done")
+
+    async def offset_xy(self, x, y, persistent=False):
+        """ Offset telescope in x and y.
+
+        Parameters
+        ----------
+        x : `float`
+            Offset in camera x-axis (arcsec).
+        y : `float`
+            Offset in camera y-axis (arcsec).
+        persistent : `bool`
+            User persistent offset instead (default=False)? Persistent offsets
+            are cumulative.
+        """
+
+        self.log.debug(f"Applying x/y offset: {x}/ {y} ")
+        azel = await self.rem.atmcs.tel_mount_AzEl_Encoders.aget()
+        nasmyth = await self.rem.atmcs.tel_mount_Nasmyth_Encoders.aget()
+        angle = (
+            np.mean(azel.elevationCalculatedAngle)
+            - np.mean(nasmyth.nasmyth2CalculatedAngle)
+            + 90.0
+        )
+        el, az, _ = np.matmul([x, y, 0.0], self.rotation_matrix(angle))
+
+        await self.offset_azel(az, el, persistent)
+
+    async def wait_for_inposition(self, timeout, cmd_ack=None, wait_settle=True):
+        """Wait for both the ATMCS and ATDome to be in position.
+
+        Parameters
+        ----------
+        timeout: `float`
+            How long should it wait before timing out.
+
+        Returns
+        -------
+        status: `str`
+            String with final status.
+        """
+        status = list()
+
+        if self.check.atmcs:
+            status.append(
+                await self.wait_for_atmcs_inposition(timeout, cmd_ack, wait_settle)
+            )
+
+        if self.check.atdome:
+            status.append(await self.wait_for_atdome_inposition(timeout, cmd_ack))
+
+        return f"{status!r}"
+
+    async def wait_for_atmcs_inposition(self, timeout, cmd_ack=None, wait_settle=True):
+        """Wait for inPosition of atmcs to be ready.
+
+        Parameters
+        ----------
+        timeout: `float`
+            How long should it wait before timing out.
+
+        Returns
+        -------
+        status: `str`
+            String with final status.
+
+        Raises
+        ------
+        asyncio.TimeoutError
+            If does not get a status update in less then `timeout` seconds.
+        """
+        while True:
+
+            try:
+                in_position = await self.rem.atmcs.evt_allAxesInPosition.next(
+                    flush=False, timeout=timeout
+                )
```

Comment:
> You are getting the oldest queued value. If you start a slew and then call this, and the telescope and dome were in position before the slew, this will return right away. I suspect that's not what you want.

Actually that is exactly what I wanted...

Reply:
Let me rephrase my concern:
Suppose the telescope is slewed several times (e.g. by somebody else or a different script) before this code has a chance to run. That will lead to several instances of "in position" and "not in position" being on the queue for this event topic. When you await `next(flush=False, ...)` you get the oldest one value, which may be quite out of date. And if you call repeatedly it may take several reads (up to 100 if there has been enough activity) before you actually get information that is recent enough to be relevant.